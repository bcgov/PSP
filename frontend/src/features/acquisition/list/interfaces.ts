export interface IAcquisitionFilter {
  acquisitionFileStatusTypeCode: string;
  acquisitionFileNameOrNumber: string;
  projectNameOrNumber: string;
}
