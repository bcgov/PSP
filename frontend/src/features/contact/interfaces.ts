export interface IContactFilter {
  summary: string;
  municipality: string;
  activeContactsOnly: boolean;
  searchBy: string;
}
