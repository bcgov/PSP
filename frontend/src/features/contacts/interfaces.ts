export interface IContactFilter {
  summary: string;
  municipality: string;
  activeContactsOnly: boolean;
  searchBy: string;
}

export enum ContactTypes {
  ORGANIZATION = 'O',
  INDIVIDUAL = 'P',
}
