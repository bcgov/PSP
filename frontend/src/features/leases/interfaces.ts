export interface ILeaseFilter {
  pinOrPid: string;
  lFileNo: string;
  searchBy: string;
  tenantName: string;
  programs: string[];
}
