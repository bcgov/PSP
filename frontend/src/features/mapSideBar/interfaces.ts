import { AddressTypes, PropertyTypes } from 'constants/index';
import { IAddress } from 'interfaces';

export const defaultPropertyValues = {
  address: {
    addressTypeId: AddressTypes.Physical,
    streetAddress1: '',
    municipality: '',
    postal: '',
    province: '',
    provinceId: 1,
    region: '',
    district: '',
  } as IAddress,
  pid: '',
  propertyTypeId: PropertyTypes.Land,
  classification: '',
  region: '',
  district: '',
  zoning: '',
  landArea: '',
  landLegalDescription: '',

  // TODO: Theses properties probably should be replaced or updated.
  titleNumber: '',
  purpose: '',
  ruralArea: '',
  ocpDesignation: '',
};
