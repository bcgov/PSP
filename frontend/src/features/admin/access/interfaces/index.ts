import { AccessRequestStatus } from 'constants/accessStatus';

export interface IAccessRequestModel {
  id: number;
  userId: number;
  businessIdentifierValue: string;
  firstName: string;
  surname: string;
  email: string;
  position: string;
  status: AccessRequestStatus;
  organization: string;
  role: string;
  note: string;
}
