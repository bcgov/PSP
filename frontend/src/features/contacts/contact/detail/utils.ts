import { IContactAddress } from 'interfaces/IContact';

import { AddressTypes } from './../../../../constants/addressTypes';

// the order of this array corresponds to the expected display order
const addressSortOrder = [AddressTypes.Mailing, AddressTypes.Residential, AddressTypes.Billing];

export const sortAddresses = (a1: IContactAddress, a2: IContactAddress) => {
  if (a2.addressType.id === a1.addressType.id) {
    return 0;
  }
  const a2Index = addressSortOrder.indexOf(a2.addressType.id as AddressTypes);
  if (a2Index !== -1 && a2Index < addressSortOrder.indexOf(a1.addressType.id as AddressTypes)) {
    return 1;
  }

  return -1;
};

export const fakeAddresses = [
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'BILLING',
      description: 'Billing address',
      isDisabled: false,
    },
    streetAddress1: 'Billing address',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6A 5G7',
  },
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'MAILING',
      description: 'Mailing address',
      isDisabled: false,
    },
    streetAddress1: 'Mailing address',
    streetAddress2: 'Living in a van',
    streetAddress3: 'Down by the River',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6Z 5G7',
  },
  {
    id: 5,
    rowVersion: 1,
    addressType: {
      id: 'RESIDNT',
      description: 'Property address',
      isDisabled: false,
    },
    streetAddress1: 'Property address',
    streetAddress2: '',
    streetAddress3: '',
    municipality: 'vic',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V9A 0H6',
  },
];
