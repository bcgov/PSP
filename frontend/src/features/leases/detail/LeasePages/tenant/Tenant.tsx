import { FormSection } from 'components/common/form/styles';
import * as Styled from 'features/leases/detail/styles';
import { FieldArray, getIn, useFormikContext } from 'formik';
import { ILease, IOrganization, IPerson } from 'interfaces';
import * as React from 'react';
import styled from 'styled-components';
import { withNameSpace } from 'utils/formUtils';

import TenantOrganizationContactInfo from './TenantOrganizationContactInfo';
import TenantPersonContactInfo from './TenantPersonContactInfo';

export interface ITenantProps {
  nameSpace?: string;
}

/**
 * Tenant lease page displays all tenant information (persons and organizations)
 * @param {ITenantProps} param0
 */
export const Tenant: React.FunctionComponent<ITenantProps> = ({ nameSpace }) => {
  const { values } = useFormikContext<ILease>();
  const persons: IPerson[] = getIn(values, withNameSpace(nameSpace, 'persons')) ?? [];
  const organizations: IOrganization[] =
    getIn(values, withNameSpace(nameSpace, 'organizations')) ?? [];

  return (
    <StyledDetails>
      <TenantsFieldArray
        name={withNameSpace(nameSpace, 'properties')}
        render={renderProps => (
          <>
            {persons.map((person: IPerson, index) => (
              <Styled.SpacedInlineListItem>
                <FormSection>
                  <TenantPersonContactInfo
                    disabled={true}
                    nameSpace={withNameSpace(nameSpace, `persons.${index}`)}
                  />
                </FormSection>
              </Styled.SpacedInlineListItem>
            ))}
            {organizations.map((organization: IOrganization, index) => (
              <Styled.SpacedInlineListItem>
                <FormSection>
                  <TenantOrganizationContactInfo
                    disabled={true}
                    nameSpace={withNameSpace(nameSpace, `organizations.${index}`)}
                  />
                </FormSection>
              </Styled.SpacedInlineListItem>
            ))}
          </>
        )}
      />
    </StyledDetails>
  );
};

export const TenantsFieldArray = styled(FieldArray)`
  column-count: 2;
  & > * {
    break-inside: avoid-column;
  }
  column-gap: 10rem;
  li {
    list-style-type: none;
    padding: 2rem 0;
    margin: 0;
  }
  @media only screen and (max-width: 1500px) {
    column-count: 1;
  }
`;

const StyledDetails = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  gap: 2.5rem;
`;

export default Tenant;
