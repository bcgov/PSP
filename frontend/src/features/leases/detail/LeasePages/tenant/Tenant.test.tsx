import { Formik } from 'formik';
import { createMemoryHistory } from 'history';
import { defaultFormLease, IFormLease } from 'interfaces';
import { noop } from 'lodash';
import { mockOrganization, mockPerson } from 'mocks/filterDataMock';
import { render, RenderOptions } from 'utils/test-utils';

import Tenant, { ITenantProps } from './Tenant';

const history = createMemoryHistory();

describe('Tenant component', () => {
  const setup = (renderOptions: RenderOptions & ITenantProps & { lease?: IFormLease } = {}) => {
    // render component under test
    const component = render(
      <Formik onSubmit={noop} initialValues={renderOptions.lease ?? defaultFormLease}>
        <Tenant nameSpace={renderOptions.nameSpace} />
      </Formik>,
      {
        ...renderOptions,
        history,
      },
    );

    return {
      component,
    };
  };
  it('renders as expected', () => {
    const { component } = setup({
      lease: { ...defaultFormLease, persons: [mockPerson], organizations: [mockOrganization] },
    });
    expect(component.asFragment()).toMatchSnapshot();
  });
  it('renders one Person Tenant section per person', () => {
    const { component } = setup({
      lease: { ...defaultFormLease, persons: [mockPerson, mockPerson] },
    });
    const { getAllByText } = component;
    const personTenant = getAllByText('Tenant Name:');

    expect(personTenant).toHaveLength(2);
  });

  it('renders one notes section per tenant note', () => {
    const { component } = setup({
      lease: { ...defaultFormLease, tenantNotes: ['note one', 'note two'] },
    });
    const { getAllByText } = component;
    const personTenant = getAllByText('Notes:');

    expect(personTenant).toHaveLength(2);
  });

  it('renders no person information section if there are no persons', () => {
    const { component } = setup({
      lease: { ...defaultFormLease, persons: [] },
    });
    const { queryByText } = component;
    const personTenant = queryByText('Tenant Name:');

    expect(personTenant).toBeNull();
  });

  it('renders one organization tenant section per organization', () => {
    const { component } = setup({
      lease: { ...defaultFormLease, organizations: [mockOrganization, mockOrganization] },
    });
    const { getAllByText } = component;
    const organizationTenant = getAllByText('Tenant organization:');

    expect(organizationTenant).toHaveLength(2);
  });

  it('renders no organization information section if there are no organizations', () => {
    const { component } = setup({
      lease: { ...defaultFormLease, organizations: [] },
    });
    const { queryByText } = component;
    const organizationTenant = queryByText('Tenant organization:');

    expect(organizationTenant).toBeNull();
  });

  it('renders organization phone numbers as expected', () => {
    const { component } = setup({
      lease: { ...defaultFormLease, organizations: [mockOrganization] },
    });
    const { getByLabelText } = component;
    const landline = getByLabelText('Landline:');
    const mobile = getByLabelText('Mobile:');

    expect(landline).toHaveDisplayValue('222-333-4444');
    expect(mobile).toHaveDisplayValue('555-666-7777');
  });

  it('renders person phone numbers as expected', () => {
    const { component } = setup({
      lease: { ...defaultFormLease, persons: [mockPerson] },
    });
    const { getByLabelText } = component;
    const landline = getByLabelText('Landline:');
    const mobile = getByLabelText('Mobile:');

    expect(landline).toHaveDisplayValue('222-333-4444');
    expect(mobile).toHaveDisplayValue('555-666-7777');
  });
});
