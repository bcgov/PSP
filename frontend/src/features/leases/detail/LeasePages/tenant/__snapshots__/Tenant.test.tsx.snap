// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tenant component renders as expected 1`] = `
<DocumentFragment>
  @media only screen and (max-width:1500px) {

}

@media only screen and (max-width:1500px) {

}

<div
    class="Toastify"
  />
  @media only screen and (max-width:1500px) {

}

@media only screen and (max-width:1500px) {

}

<div
    class="Toastify"
  />
  .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  gap: 0.8rem;
}

.c5 .form-label {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 {
  border-radius: 1rem;
  width: 100%;
  padding: 2.5rem;
}

.c4 {
  font-size: 2rem;
  margin-bottom: 1rem;
  text-align: left;
  padding: 1rem 0 0.5rem 0;
  border-bottom: solid 0.3rem;
}

.c3 {
  display: grid;
  grid-template-columns: [labels] minMax(min-content,1fr) [controls] 5fr;
  grid-auto-flow: row;
}

.c3 .form-label,
.c3 .form-group {
  margin: 0 0.5rem 0 0;
}

.c3 > .input {
  grid-column: controls;
  grid-row: auto;
  border-left: 1px solid black;
}

.c3 .form-control {
  font-weight: 700;
}

.c3 > label,
.c3 > fieldset {
  grid-column: labels;
  grid-row: auto;
}

.c3 > .textarea,
.c3 > h3 {
  grid-column: span 2;
  border-left: none;
}

.c3 .form-label {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-column-gap: 5rem;
  column-gap: 5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c6 {
  padding: 0.6rem 1.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 .form-label {
  min-width: 7rem;
}

.c1 {
  -webkit-column-count: 2;
  column-count: 2;
  -webkit-column-gap: 10rem;
  column-gap: 10rem;
}

.c1 > * {
  -webkit-break-inside: avoid-column;
  break-inside: avoid-column;
}

.c1 li {
  list-style-type: none;
  padding: 2rem 0;
  margin: 0;
}

@media only screen and (max-width:1500px) {
  .c1 {
    -webkit-column-count: 1;
    column-count: 1;
  }
}

@media only screen and (max-width:1500px) {

}

<div
    class="c0 c1"
  >
    <li
      class="c2"
    >
      <div
        class="c0"
      >
        <div
          class="c3"
        >
          <label
            class="form-label"
          >
            Tenant Name:
          </label>
          <div
            class="input form-group"
          >
            <input
              class="form-control"
              disabled=""
              id="input-persons.0.fullName"
              name="persons.0.fullName"
              value=""
            />
          </div>
          <h3
            class="c4"
          >
            Contact Info
          </h3>
          <label
            class="form-label"
          >
            Address:
          </label>
          <div
            class="input form-group"
          >
            <input
              class="form-control"
              disabled=""
              id="input-persons.0.address.province"
              name="persons.0.address.province"
              value=""
            />
          </div>
          <br />
          <label
            class="form-label"
          >
            E-mail address:
          </label>
          <div
            class="input form-group"
          >
            <input
              class="form-control"
              disabled=""
              id="input-persons.0.email"
              name="persons.0.email"
              value="admin@pims.gov.bc.ca"
            />
          </div>
          <label
            class="form-label"
          >
            Phone:
          </label>
          <div
            class="c5 c6 input form-group"
          >
            <label
              class="form-label"
              for="input-persons.0.landline"
            >
              Landline: 
            </label>
            <input
              class="form-control"
              disabled=""
              id="input-persons.0.landline"
              name="persons.0.landline"
              value="1 222-333-4444"
            />
          </div>
          <div
            class="c5 c6 input form-group"
          >
            <label
              class="form-label"
              for="input-persons.0.mobile"
            >
              Mobile: 
            </label>
            <input
              class="form-control"
              disabled=""
              id="input-persons.0.mobile"
              name="persons.0.mobile"
              value="1 555-666-7777"
            />
          </div>
        </div>
      </div>
    </li>
    <li
      class="c2"
    >
      <div
        class="c0"
      >
        <div
          class="c3"
        >
          <label
            class="form-label"
          >
            Tenant organization:
          </label>
          <div
            class="input form-group"
          >
            <input
              class="form-control"
              disabled=""
              id="input-organizations.0.name"
              name="organizations.0.name"
              value="Ministry of Transportation & Infrastructure"
            />
          </div>
          <label
            class="form-label"
          >
            Contact name:
          </label>
          <div
            class="input form-group"
          >
            <input
              class="form-control"
              disabled=""
              id="input-organizations.0.contactName"
              name="organizations.0.contactName"
              value=""
            />
          </div>
          <h3
            class="c4"
          >
            Contact Info
          </h3>
          <label
            class="form-label"
          >
            Address:
          </label>
          <div
            class="input form-group"
          >
            <input
              class="form-control"
              disabled=""
              id="input-organizations.0.address.streetAddress1"
              name="organizations.0.address.streetAddress1"
              value="1234 mock Street"
            />
          </div>
          <div
            class="input form-group"
          >
            <input
              class="form-control"
              disabled=""
              id="input-organizations.0.address.streetAddress2"
              name="organizations.0.address.streetAddress2"
              value="N/A"
            />
          </div>
          <div
            class="input form-group"
          >
            <input
              class="form-control"
              disabled=""
              id="input-organizations.0.address.municipality"
              name="organizations.0.address.municipality"
              value="Victoria"
            />
          </div>
          <div
            class="input form-group"
          >
            <input
              class="form-control"
              disabled=""
              id="input-organizations.0.address.postal"
              name="organizations.0.address.postal"
              value="V1V1V1"
            />
          </div>
          <div
            class="input form-group"
          >
            <input
              class="form-control"
              disabled=""
              id="input-organizations.0.address.province"
              name="organizations.0.address.province"
              value="BC"
            />
          </div>
          <br />
          <label
            class="form-label"
          >
            E-mail address:
          </label>
          <div
            class="input form-group"
          >
            <input
              class="form-control"
              disabled=""
              id="input-organizations.0.email"
              name="organizations.0.email"
              value=""
            />
          </div>
          <label
            class="form-label"
          >
            Phone:
          </label>
          <div
            class="c5 c6 input form-group"
          >
            <label
              class="form-label"
              for="input-organizations.0.landline"
            >
              Landline: 
            </label>
            <input
              class="form-control"
              disabled=""
              id="input-organizations.0.landline"
              name="organizations.0.landline"
              value="1 222-333-4444"
            />
          </div>
          <div
            class="c5 c6 input form-group"
          >
            <label
              class="form-label"
              for="input-organizations.0.mobile"
            >
              Mobile: 
            </label>
            <input
              class="form-control"
              disabled=""
              id="input-organizations.0.mobile"
              name="organizations.0.mobile"
              value="1 555-666-7777"
            />
          </div>
        </div>
      </div>
    </li>
  </div>
</DocumentFragment>
`;
